unity/Assets/Scripts/Util/Geometry/Contour/ContourPolygon.cs

```diff
@@ -0,0 +1,236 @@
+    ///  - For every parent contour its children contours are listed.
+    ///  - This results in a tree structure with levels ccw - cw - ccw - cw etc.
+    ///
+    /// All ContourPolygons created should adhere to this specification. If not, operations might return wrong results,
+    /// e.g. returning a negative area.
+    /// </summary>
+    public class ContourPolygon : IPolygon2D
+    {
+        /// <summary>
+        /// List of contours that define the polygon. There can be multiple external contours and multiple
+        /// internal contours. Internal contours (holes) are stored by using their ID within the Contour.
+        /// </summary>
+        public List<Contour> Contours { get; private set; }
+
+        public int NumberOfContours
+        {
+            get { return Contours.Count; }
+        }
+
+        public int VertexCount
+        {
+            get { return Contours.Aggregate(0, (acc, contour) => acc + contour.VertexCount); }
+        }
+        
+        public ICollection<Vector2> Vertices
+        {
+            get { return Contours.SelectMany(v => v.Vertices).ToList(); }
+        }
+        
+        public ICollection<LineSegment> Segments
+        {
+            get { return Contours.SelectMany(v => v.Segments).ToList(); }
+        }
+
+        public ContourPolygon()
+        {
+            Contours = new List<Contour>();
+        }
+
+        public ContourPolygon(IEnumerable<Contour> contours)
+        {
+            Contours = new List<Contour>(contours);
+        }
+
+        public void Add(Contour c)
+        {
+            Contours.Add(c);
+        }
+        
+        public Contour this[int index]
+        {
+            get
+            {
+                return Contours[index];
+            }
+        }
+
+        public void Join(ContourPolygon pol)
+        {
+            var size = NumberOfContours;
+            Contours.AddRange(pol.Contours);
+            foreach (var contour in pol.Contours)
+            {
+                contour.ClearHoles();
+                foreach (var holeId in contour.Holes)
+                {
+                    Contours.Last().AddHole(holeId + size);
+                }
+            }
+        }
+
+        public Rect BoundingBox(float margin = 0f)
+        {
+            return BoundingBoxComputer.FromPoints(Vertices, margin);
+        }
+
+        public float Area
+        {
+            get { return Contours.Aggregate(0f, (acc, contour) => acc + contour.Area); }
+        }
+
+        public string Visualize()
+        {
+            var result = new StringBuilder();
+            foreach (var contour in Contours)
+            {
+                result.Append("\\draw ");
+                foreach (var point in contour.Vertices)
+                {
+                    result.AppendFormat("({0:f2}, {1:f2}) -- ", point.x, point.y);
+                }
+
+                result.Append("cycle;");
+            }
+
+            return result.ToString();
+        }
```

Comment:
Is this code to generate a image form the polygon in latex? That might be handy for debugging. Can we also implement this for the normal polygon?

Reply:
Yes, this was added for latex. Adding it to `Polygon2D` is probably easier because you can remove the outer loop and only need to create the inner loop.